### DISCUSSION TITLE
Push all from desktop to online?

### DISCUSSION THREAD

**Discussion by thierry-laval** [community/community#23719] [New to GitHub]
<p>324/5000</p>
<p>Hello, In github desktop I have several local files (800 files) that I would like to push to my main branch in github. I find to send them one after the other but I can’t find how to send everything at once … Is this possible? And how can I do it? Thank you for your help.</p>

**Comment by SethClydesdale** [comment, marked as answer]
<p>Hello <a class="mention" href="/thierry-laval">@thierry-laval</a>, welcome to the GitHub Support Community!</p>
<p>You said you’re using GitHub Desktop, correct? To add multiple files:</p>
<ol>
<li>Copy all of the files into the locally cloned repo.</li>
<li>Commit all of the files with a summary. (a summary is required when multiple files have changed)</li>
<li>Push the changes to GitHub.</li>
</ol>
<p>Depending on the size of the files, it could take a little while to complete the task. Also, if you need any help with the steps above, please <a href="https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop" rel="nofollow noopener">see this page</a> on contributing with GitHub Desktop.</p>

**Comment by thierry-laval** [comment]
<p>On 3, yes but how ?</p>
<p>When I want to commit all my files I have to commit them one after one first! I have 800 files! How to commit directly with ALL my files? Without description… :slight_smile:</p>

**Comment by thierry-laval** [comment]
<p>Yes yes, i found my question… How to commit 800 files in one time?</p>

**Comment by SethClydesdale** [comment]
<p>Just to clarify: are you using <a href="https://desktop.github.com/" rel="nofollow noopener">GitHub Desktop</a> or the command line?</p>

**Comment by thierry-laval** [comment]
<p>I use Github desktop. (i don’t use command line)<br>
I have 800 files to commit, before push.<br>
I don’t have to write something on the files.<br>
Actualy i have to commit files one by one… So long<br>
Do you have a solution to commit all 800 files in one time ?<br>
:wink:<br>
Thank a lot for your help</p>

**Comment by SethClydesdale** [comment]
<p>Hmm… that’s the odd thing, the steps I provided above should do the trick. Are you copying all of these 800 files into your repository’s directory at once? For example, I’ve copied the folder “lessons” (which has 800+ files) into my GitHub Page’s repo. These files show up in the changes column for review.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6722bfb298e2b64f7fbfed3ce98096aef930efca.png" data-download-href="/uploads/short-url/eInCXN0qj8IFwZFiPgq0zAtUucq.png?dl=1" title="repo" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10122549/181078900-673206d0-c86d-4f68-8830-683b7402b680.png" alt="repo" data-base62-sha1="eInCXN0qj8IFwZFiPgq0zAtUucq" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/6722bfb298e2b64f7fbfed3ce98096aef930efca_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/2X/6/6722bfb298e2b64f7fbfed3ce98096aef930efca_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/6/6722bfb298e2b64f7fbfed3ce98096aef930efca_2_1380x724.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6722bfb298e2b64f7fbfed3ce98096aef930efca_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">repo</span><span class="informations">1441×758 74.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>To commit the files, I need only include a summary and then I can commit them, and finally push the changes to GitHub. To commit all the files at once, make sure to click the checkbox next to “N changed files” until every file is checked.</p>
<p>See <a href="https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/committing-and-reviewing-changes-to-your-project" rel="nofollow noopener">Committing and reviewing changes to your project</a> for more information.</p>

**Comment by thierry-laval** [comment]
<p>Hello, thank you very much for this help.<br>
It’s just awesome and very sweet.<br>
I followed your recommendations and found my (beginner) mistake.<br>
I only needed to include a summary, then I was able to commit them, and finally push the changes to GitHub.<br>
You just had to put this summary … “Summary required” to be able to push all the files at the same time.<br>
A big big thank you.<br>
If you had a site to learn all these basics?<br>
Very good day…</p>

**Comment by SethClydesdale** [comment]
<p>You’re welcome! Yes, it is indeed something easy to miss. It’s optional for single file changes, however, when committing multiple files you need to include a summary. Went ahead and updated my earlier post with that info.</p>
<p>As for learning all the basics for GitHub Desktop, you may want to look through the documentation below as it should include everything you need to know.</p>
<p><a href="https://docs.github.com/en/desktop" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/desktop</a></p>

**Comment by trextozyne** [comment]
<p>Thanks I have been having issues with this where I had to discard all changes and my apps crashed. Thanks</p>

**Comment by Sanjeev9910** [comment]
<p>Make changes in your project using your IDE/text editor.<br>
Switch back to GitHub Desktop – you should see changed files in the Changes tab in the left sidebar.<br>
Commit those changes in the bottom-left corner of GitHub Desktop. …<br>
Click the Push origin button to push the changes up to the remote repository on GitHub.</p>
